devel/smake/Portfile

```diff
@@ -1,41 +1,96 @@
+name                    smake
+version                 1.7-${codeberg.version}
+revision                0
+categories              devel
+platforms               darwin
+maintainers             {hotmail.com:amtor @RobK88} \
+                        openmaintainer
+license                 CDDL
+description             A highly portable make program with automake features
+long_description        Smake is a highly portable make program with automake \
+                        features. It is currently mainly targeted to be used \
+                        with the Schily SING makefile system.
 
-use_configure	no
+homepage                https://codeberg.org/schilytools/schilytools
+master_sites            ${homepage}/archive/
+checksums               rmd160  e154278ecbe7d778bc1d6766ed163c9963b1cc82 \
+                        sha256  a4270cdcca5dd69c0114079277b06e5efad260b0a099c9c09d31e16e99a23ff5 \
+                        size    5896292
 
-# https://trac.macports.org/ticket/31616
-compiler.blacklist-append llvm-gcc-4.2 {clang < 318}
-
-build.type		gnu
-build.args      CC=${configure.cc}
-use_parallel_build  no
-
-destroot.args       CC=${configure.cc}
-destroot.destdir	INS_BASE=${destroot}${prefix} \
-					MANDIR=man \
-					INSUSR=`id -u` INSGRP=`id -g`
-post-destroot	{
-	file delete -force ${destroot}${prefix}/share/man/man5 \
-		${destroot}${prefix}/lib ${destroot}${prefix}/include
+post-extract {
+                        copy ${filespath}/Gmake.smake ${worksrcpath}
 }
 
-livecheck.url       https://sourceforge.net/projects/s-make/files/
-livecheck.regex     ${name}-(\\d+(?:\\.\\d+)*)${extract.suffix}
+patchfiles              001-Makefile-Patch.diff
+
+post-patch {
+                        reinplace -locale C "s|/opt/schily|${prefix}|g" \
+                        ${worksrcpath}/DEFAULTS/Defaults.darwin \
+                        ${worksrcpath}/DEFAULTS/Defaults.mac-os10 \
+                        ${worksrcpath}/DEFAULTS_ENG/Defaults.darwin \
+                        ${worksrcpath}/DEFAULTS_ENG/Defaults.mac-os10 \
+                        ${worksrcpath}/libfind/find.c \
+                        ${worksrcpath}/libfind/find_main.c \
+                        ${worksrcpath}/librscg/scsi-remote.c \
+                        ${worksrcpath}/patch/tests/random/cmptest.sh \
+                        ${worksrcpath}/patch/tests/random/gentest.sh \
+                        ${worksrcpath}/smake/job.c \
+                        ${worksrcpath}/smake/make.c \
+                        ${worksrcpath}/smake/smake.1 \
+                        ${worksrcpath}/TEMPLATES/Defaults.gcc \
+                        ${worksrcpath}/TEMPLATES/Defaults.clang \
+                        ${worksrcpath}/TEMPLATES/Defaults.xcc
+}
+
+conflicts_build-append  cdrtools
+
+# https://trac.macports.org/ticket/31616
+compiler.blacklist      llvm-gcc-4.2 macports-llvm-gcc-4.2 {clang < 300}
+
+configure.ldflags-append -lintl
+
+universal_variant       no
```

Comment:
```suggestion
```

Any reason to keep it? I've successfully build universal port on 10.6 which I've used for testing.

Reply:
> Any reason to keep it? I've successfully build universal port on 10.6 which I've used for testing.

Agreed, we shouldn't drop Universal support.

Reply:
I will try again building a universal variant.  When I tried building a universal binary of `smake` a few days ago, it would NOT build one on my Mac running High Sierra.  The build did not fail but the resulting binary was only 64 bit.  I will try again.  If the port builds a universal binary, I will drop `universal_variant       no`.

Reply:
> I will try again building a universal variant. When I tried building a universal binary of `smake` a few days ago, it would NOT build one on my Mac running High Sierra. The build did not fail but the resulting binary was only 64 bit. I will try again. If the port builds a universal binary, I will drop `universal_variant no`.

Given that a bootstrap version of `smake` has to be built, in order to build the rest of the port... it may require some work to support a Universal variant. So if it's not something easy to do, don't worry about it at the moment. (I know you're trying to wrap this up ASAP, so feel free to defer this until after you come back from your trip.)

Reply:
@RobK88 you may try to add:
```tcl
PortGroup           muniversal 1.1
```
that should fix universal build if it doesn't work out of box :)

Reply:
Nope.  I cannot build the +universal variant of `smake` on my Mac running High Sierra.  

```
$ file `which smake`
/opt/local/bin/smake: Mach-O 64-bit executable x86_64
```

The build does not fail.  But the resulting binary is only 64-bit.  I suspect it is due to limitations of `smake` itself.  Remember this port creates a bootstrapped version of `smake` using `gmake`.  (The bootstrapped version of `smake` is likely universal).  But then the port creates the final version of `smake` using the bootstrapped version of `smake`.  The resulting binary is 64 bit only.  

Please looks at the other ports that use `smake` as a build dependency, such as `cdrtools` and `star`.  All these ports have `universal_variant  no` in their Portfiles.

Reply:
@RobK88 have you tried to use muniversal PG?

Reply:
Not yet.  I will try now and I will let you know.

Reply:
What is quite surprising that parallels build brokes universal build via munivesral PG.